code/modules/cards/cards.dm

```diff
@@ -0,0 +1,50 @@
+/*
+** A Deck of Cards for playing various games of chance
+*/
+/obj/item/toy/cards
+	resistance_flags = FLAMMABLE
+	max_integrity = 50
+	/// Do all the cards drop out of the deck when thrown
+	var/can_play_52_card_pickup = TRUE
+	/// List of cards for a hand or deck
+	var/list/cards = list()
+
+/obj/item/toy/cards/throw_impact(mob/living/target, datum/thrownthing/throwingdatum)
+	return
+
+/**
+/obj/item/toy/cards/suicide_act(mob/living/carbon/user)
+	user.visible_message(span_suicide("[user] is slitting [user.p_their()] wrists with \the [src]! It looks like [user.p_they()] [user.p_have()] a crummy hand!"))
+	playsound(src, 'sound/items/cardshuffle.ogg', 50, TRUE)
+	return BRUTELOSS
+**/
+
+/obj/item/toy/cards/proc/insert(list/cards_to_add)
+	for(var/obj/item/toy/singlecard/card in cards_to_add)
+		card.forceMove(src)
+		cards += card
+	update_appearance()
+
+/**
+ * draw
+ *
+ * Draws a card from the deck or hand of cards.
+ *
+ * Arguments:
+ * * mob/living/user - The user drawing the card.
+ * * obj/item/toy/singlecard/card - The card drawn from the hand
+**/ 
+/obj/item/toy/cards/proc/draw(mob/living/user, obj/item/toy/singlecard/card)
+	if(!isliving(user) || !user.canUseTopic(src, BE_CLOSE, NO_DEXTERITY, NO_TK))
+		return
+
+	var/has_no_cards = !LAZYLEN(cards)
+	if(has_no_cards)
```

Comment:
does this even need to be a var?

Reply:
I would highly recommend doing this for easier readability when you are programming.  It's more human readable:

```
if(card_was_inserted && deck_is_not_empty)
```

vs

```
if(src.cards[card] && parent_deck.len > 1)
```

Reply:
bottom is readable, considering you don't need to use `src.` there

Reply:
It was a pseudocode example.  I can read DM code just fine most of the time.  Breaking down code logic into human readable chunks is ideal especially considering the size of the repo and the amount of non-coder contributors.  I think a lot of programmers develop tunnel vision when they become proficient in a language and forget what it's like starting out.

![df_meme](https://user-images.githubusercontent.com/5195984/156361530-fa3f58c0-9fd3-4248-9ae1-cd2ba2bda090.jpg)

To someone in Python the above code might be written as:

```python
has_no_cards = not cards
OR
has_no_cards = len(cards) == 0
```

Or using Lua it would be:

```lua
local has_no_cards = (next(cards) == nil)
```

Regardless of the code implementation `has_no_cards` is human readable by any coder or noncoder.  The only thing we sacrifice is a tiny breadcrumb of memory to improve code readability.  You shouldn't do this where memory might become a big problem or in a performance critical piece of code, but everywhere else is fine.

Reply:
i get the point but its simple enough that it doesn't really matter